lib/zadarest/client.rb

```diff
@@ -0,0 +1,56 @@
+module Zadarest
+  class Client
+    def initialize(config: nil)
+      @config = config
+      @client = Faraday.new(url: config.api_endpoint) do |f|
+        f.request :url_encoded
+        f.adapter Faraday.default_adapter
+      end
+    end
+
+    def post(url: '', params: {}, root_key: 'response')
```

Comment:
These methods are dense but there's still some duplication. Think about how to eliminate it.